File: 수학(구현)/소수 찾기(배열X)/백준_4948_베르트랑 공준/베르트랑 공준.java
import java.util.*;

public class Main {
    static Boolean is_prime(int n) {
        for (int i = 2; i * i <= n; i++) {
            if (n % i == 0)
                return false;
        }
        return true;
    }

    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);

        int n;
        while (true) {
            n = s.nextInt();
            int count = 0;
            if (n == 0)
                break;

            for (int i = n + 1; i <= 2 * n; i++) {
                if (is_prime(i))
                    count++;
            }
            System.out.println(count);
        }
    }
}

File: 수학(구현)/소수 찾기(배열X)/백준_4948_베르트랑 공준/베르트랑 공준.py
def is_prime(k):
    for i in range(2, k+1):
        if i*i > k:
            break
        if k % i == 0:
            return False
    return True

while(True):
    n = int(input())
    if n == 0:
        break
    count = 0

    for i in range(n+1, 2*n+1):
        if is_prime(i):
            count += 1
    print(count)